File: LeetCode/1582/main.py
class Solution:
    def numSpecial(self, mat: List[List[int]]) -> int:
        m, n = len(mat), len(mat[0])
        rows, cols = [0]*m, [0]*n
        ans = 0

        for i in range(m) :
            for j in range(n) :
                rows[i] += mat[i][j]
                cols[j] += mat[i][j]
        
        for i in range(m) :
            for j in range(n) :
                ans += (mat[i][j] == 1 and rows[i] == 1 and cols[j] == 1)
        
        return ans

File: LeetCode/1582/main.cpp
class Solution {
public:
    int numSpecial(vector<vector<int>>& mat) {
        int m = mat.size();
        int n = mat[0].size();
        int ans = 0;
        vector<int> rows(m), cols(n);

        for(int i = 0; i < m; i++) {
            for(int j = 0; j < n; j++) {
                rows[i] += mat[i][j];
                cols[j] += mat[i][j];
            }
        }
        for(int i = 0; i < m; i++) {
            for(int j = 0; j < n; j++) 
                ans += (rows[i] == 1 && cols[j] == 1 && mat[i][j] == 1);
        }
        return ans;
    }
};